File: api/Models/Sale.cs
public class Sale
{
    public int Id { get; set; }
    public int CarId { get; set; }
    public int OwnerId { get; set; }
    public decimal SalePrice { get; set; }
    public DateTime SaleDate { get; set; } = DateTime.UtcNow;
    public required Car Car { get; set; }
    public Owner? Owner { get; set; }
}

File: app/types/sale.ts
import { Car } from './car';
import { Owner } from './owner';

export interface Sale {
  id: number;  // Assuming you have an ID field for each sale
  car: Car;  // Embed the Car type
  owner: Owner;  // Embed the Owner type
  sale_price: number;  // Decimal value represented as a number
  sale_date: string;  // ISO string representation of the date
}
